IQSS/dataverse-frontend
# Feature/319 new collection page

**What this PR does / why we need it**:
Adds section/page to create a new collection and integrates with the js-dataverse use case.

**Which issue(s) this PR closes**:

Closes #319 

**Special notes for your reviewer**:

- This PR does not contemplate checking whether or not the user can create a collection within a non-root collection, nor if the user is authenticated or not.
- It only sends required fields since the use case only contemplates required fields to create a collection for now. 
- A feature has been added to suggest an identifier to the user based on the collection name entered.

**Suggestions on how to test this**:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the New Collection page**
1. Log in.
2. From the root collection page click on Add Data => New Collection button.
3. Check that Host Collection, Collection Name, Affiliation and Email fields are correctly pre-filled with your account data.
4. Apply the suggestion about the Identifier field. Also you can change the Collection Name for something else and a new suggestion for the Identifier field will appear.
5. Select a Category.
6. Submit the Form and you will be redirected to the just created Collection page with a successful creation alert.
7. You can also try creating a collection standing in another sub collection and check that the Host Collection should be the correct one.
8. Note that clicking on Add Data => New Collection from the header will always create a Collection with Root as owner collection.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
_**Full size screenshot 👇**_
![full_screen_size](https://github.com/user-attachments/assets/ea051feb-6196-4114-9c56-424bb8b30281)

_**Identifier suggestion functionality 👇**_
https://github.com/user-attachments/assets/8d8cee29-81f2-4e9a-b6f3-8549de4d46ba

**Is there a release notes update needed for this change?**:
N/A

**Additional documentation**:
N/A
